equipmanager.lic

```diff
@@ -18,7 +26,32 @@ class EquipmentManager
     @gear_sets = {}
     settings.gear_sets.each { |set_name, gear_list| @gear_sets[set_name] = gear_list.flatten.uniq }
     @sort_head = settings.sort_auto_head
-    @items = settings.gear.map { |item| Item.new(name: item[:name], leather: item[:is_leather], hinders_locks: item[:hinders_lockpicking], worn: item[:is_worn], swappable: item[:swappable], tie_to: item[:tie_to], adjective: item[:adjective], bound: item[:bound], wield: item[:wield], transforms_to: item[:transforms_to], transform_verb: item[:transform_verb], transform_text: item[:transform_text], lodges: item[:lodges], ranged: item[:ranged], needs_unloading: item[:needs_unloading], skip_repair: item[:skip_repair], container: item[:container]) }
+    @items = settings.gear.map { |item| Item.new(name: item[:name], leather: item[:is_leather], hinders_locks: item[:hinders_lockpicking], worn: item[:is_worn], swappable: item[:swappable], tie_to: item[:tie_to], adjective: item[:adjective], bound: item[:bound], wield: item[:wield], transforms_to: item[:transforms_to], transform_verb: item[:transform_verb], transform_text: item[:transform_text], lodges: item[:lodges], ranged: item[:ranged], needs_unloading: item[:needs_unloading], skip_repair: item[:skip_repair], container: item[:container], crafts: item[:crafts], tool_types: item[:tool_types]) }
+  end
+
+  def get_tool?(craft_type, tool_type)
+    item = items.find { |item| item.crafts.include?(craft_type) && item.tool_types.include?(tool_type) }
+    return false unless item
+
+    return false unless get_item?(item)
+
+    return true unless item.swappable
+
+    return swap_to_tool_type?(item, tool_type)
+  end
+
+  def swap_to_tool_type?(item, tool_type)
+    tool_swap_command = "adjust my #{item.adjective} #{item.name}"
+
+    item.tool_types.length.times {
+      case DRC.bput(tool_swap_command, SHOVEL_MATCHES, TONGS_MATCHES)
+      when *SHOVEL_MATCHES
+        return true if tool_type =~ /shovel/i
```

Comment:
If the output matches the “when” statement, are the regex matches on tool_type for shovel or tongs still necessary? Could a tool have the right message but different noun?

Reply:
it's checking that the swap was to the tool_type you want. the tongs are both tongs and shovels, for example, so you have to swap them around till you get to what you want.

Those don't have to be regex checks against the tool_type param, they could be equality checks, etc.